### PR TITLE
Add babel and externals support in Webpack for add-on infrastructure

### PR DESCRIPTION
**Pull request by sneridagh**
So this is it.

I couldn't made it work with bare Volto, I have no clue why... but it works in a standard Volto project... I guess that's where it counts 😀😀

package.json
```json
...
"addons": [
    "volto-myaddon"
]
...
```

So it's left the config aggregator and what else?

### CODE REVIEW

**Comment by sneridagh** [comment]
okay, will check what happens later, for not defined addons.

File: razzle.config.js [hunk 1]
```diff
@@ -287,12 +287,25 @@ module.exports = {
     if (packageJson.name !== '@plone/volto') {
       include.push(fs.realpathSync(`${voltoPath}/src`));
     }
+    packageJson.addons.forEach(addon =>
+      include.push(
+        fs.realpathSync(`${projectRootPath}/node_modules/${addon}/src`),
```
Comment (sneridagh) [inline razzle.config.js]: We need to decide if we want the addon to have `src` or not.

**Comment by sneridagh** [comment]
Related: #1433

**Comment by tiberiuichim** [comment]
I like it. This opens the door to a lot of other possibilities, including automatic addon configuration.

Not directly related to this PR, it would be nice to have some references to the rules declared in razzle.config.js, so that this sort of dance:
```
    const babelRuleIndex = config.module.rules.findIndex(
      rule =>
        rule.use &&
        rule.use[0].loader &&
        rule.use[0].loader.includes('babel-loader'),
    );
``` 
is not required in Volto projects.

**Comment by nzambello** [comment]
This looks nice, I find it a good first step for an addon system likely automagical.

**Comment by sneridagh** [comment]
I made some tests, it works perfectly with the current infrastructure (at least ours, based in yarn workspaces), so it's not breaking. I will merge it, but it would be great if you can test it on your own before I cut a release.

For those using yarn workspaces, with this one, the `rm -rf ./node_modules/mypackage` workaround is no longer needed, if the `addon` entry is declared in `package.json`!

So we can have full `yarn workspaces` feature working in Volto (which I think is the way to go). It works either with mrs.developer setups (to develop add-ons while developing projects) and with released packages.

/cc @ebrehault @vangheem

**Pull request merged**